### PR TITLE
Add missing localized messages

### PR DESCRIPTION
**Pull request by mcelicalderon**
This makes sure we don't rely on Devise or DTA's locale files to render message translations. It also adds some missing translations that were used on mailer templates.

### CODE REVIEW

**Comment by aarona** [comment]
One thing I had noticed when playing around the locale stuff is that because `devise.en.yml` comes before `en.yml`, I think anything in the dummy's `devise.en.yml` will override anything you put in `en.yml`. Also, because `devise.en.yml` is included in the dummy file, that's why I was under the impression you guys wanted to fall back on those locale's (And this PR proves that was the case until now).

TL;DR do we need `devise.en.yml`? Should we delete it since when testing we don't want to rely on that file? Also we should assume the user might not have that file anyway, right? Just a suggestion.

**Comment by mcelicalderon** [comment]
You are right, we shouldn't rely on that one for the tests, I'll remove it. I think it was added when running this gem's generator which runs devise's generator..

**Comment by aarona** [comment]
Yeah this gem is getting more mature now so no need to bootstrap it with fallback code/files!
